internal/k8s/status.go

```diff
@@ -176,24 +177,14 @@ func (irs *StatusWriter) setIngressRouteStatus(existing, updated *ingressroutev1
 	return err
 }
 
-func (irs *StatusWriter) setHTTPProxyStatus(existing, updated *projcontour.HTTPProxy) error {
-	existingBytes, err := json.Marshal(existing)
-	if err != nil {
-		return err
-	}
-	// Need to set the resource version of the updated endpoints to the resource
-	// version of the current service. Otherwise, the resulting patch does not
-	// have a resource version, and the server complains.
-	updated.ResourceVersion = existing.ResourceVersion
-	updatedBytes, err := json.Marshal(updated)
-	if err != nil {
-		return err
-	}
-	patchBytes, err := jsonpatch.CreateMergePatch(existingBytes, updatedBytes)
+func (irs *StatusWriter) setHTTPProxyStatus(updated *projcontour.HTTPProxy) error {
+
+	usUpdated, err := irs.Converter.ToUnstructured(updated)
 	if err != nil {
-		return err
+		return fmt.Errorf("Unable to convert status update to HTTPPRoxy: %s", err)
```

Comment:
Nit, drop the initial caps

https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
s/HTTPPRoxy/HTTPProxy/

Reply:
One of these days I'll get that right, sigh. On both counts.